client/messaging/common-message-layout.tsx

```diff
@@ -34,55 +36,84 @@ const Text = styled.span`
   overflow: hidden;
 `
 
-export function ParsedText({ text }: { text: string }) {
-  const parsedText = useMemo(() => {
-    const matches = matchLinks(text)
+const UserMention = styled.span`
+  background-color: ${rgba(blue400, 0.48)};
+
+  &:hover {
+    background-color: ${rgba(blue400, 0.87)};
+  }
+`
+
+function* getAllMatches(text: string) {
+  yield* matchMentionsMarkup(text)
+  yield* matchLinks(text)
+}
+
+export const TextMessage = React.memo<{
+  userId: SbUserId
+  selfUserId: SbUserId
+  time: number
+  text: string
+}>(props => {
+  const { userId, selfUserId, time, text } = props
+  const [parsedText, isHighlighted] = useMemo(() => {
+    const matches = getAllMatches(text)
+    const sortedMatches = Array.from(matches).sort((a, b) => a.index - b.index)
     const elements = []
     let lastIndex = 0
+    let isHighlighted = false
 
-    for (const match of matches) {
+    for (const match of sortedMatches) {
       // Insert preceding text, if any
       if (match.index! > lastIndex) {
         elements.push(text.substring(lastIndex, match.index))
       }
 
-      elements.push(
+      if (match.type === 'mentionMarkup') {
+        const userId = Number(match.groups.userId) as SbUserId
```

Comment:
Same as in the tests, basically never do this cast. It works *now* but if the type changes or we change how we generate the opaque type, it may not. *Always* use `makeSbUserId`, it makes the intention very clear and can be updated to deal with implementation changes in one place.